diff --git a/src/sunstone/public/app/app.js b/src/sunstone/public/app/app.js
index 5f2cf79c4..50ea11da8 100644
--- a/src/sunstone/public/app/app.js
+++ b/src/sunstone/public/app/app.js
@@ -149,7 +149,7 @@ define(function(require) {
               <i class='fas fa-fw fa-check'></i>" + Locale.tr("All") + "</a></li>";
           }
 
-          if (!$.isArray(groups)){
+          if (!Array.isArray(groups)){
             groups = groups.toString();
             groups = [groups];
           }
@@ -267,7 +267,6 @@ define(function(require) {
     });
 
     $(".user-zone-info a.configuration").click(function() {
-      //$(document).foundation('dropdown', 'closeall');
       Sunstone.showTab(SETTINGS_TAB_ID);
     });
   }
diff --git a/src/sunstone/public/app/console/spice.js b/src/sunstone/public/app/console/spice.js
index 0f09255b3..e4fc04b1d 100644
--- a/src/sunstone/public/app/console/spice.js
+++ b/src/sunstone/public/app/console/spice.js
@@ -73,7 +73,6 @@ define(function(require) {
     path = spice_query_var('path', 'websockify');
 
     if ((!host) || (!port)) {
-      console.log("must specify host and port in URL");
       return;
     }
 
@@ -119,8 +118,6 @@ define(function(require) {
       spice_xfer_area.setAttribute('id', 'spice-xfer-area');
       document.getElementById('spice-area').addEventListener('dragover', handle_file_dragover, false);
       document.getElementById('spice-area').addEventListener('drop', handle_file_drop, false);
-    } else {
-      console.log("File API is not supported");
     }
   }
 
diff --git a/src/sunstone/public/app/console/vnc.js b/src/sunstone/public/app/console/vnc.js
index 8688a5e6f..5fa069584 100644
--- a/src/sunstone/public/app/console/vnc.js
+++ b/src/sunstone/public/app/console/vnc.js
@@ -170,6 +170,5 @@ define(function(require) {
     _rfb.addEventListener("credentialsrequired", credentialsRequired);
   }catch(err){
     setStatus("Something went wrong, connection is closed", "Failed");
-    console.log("error start NOVNC ", err);
   }
 });
diff --git a/src/sunstone/public/app/login.js b/src/sunstone/public/app/login.js
index 710477850..2747ede3f 100644
--- a/src/sunstone/public/app/login.js
+++ b/src/sunstone/public/app/login.js
@@ -67,7 +67,7 @@ define(function(require) {
         }
         $("#webauthn_login_btn").click(function () {
           WebAuthnJSON.get({ "publicKey": response }).then(authenticate)
-          .catch((e) => {
+          .catch(function(e) {
             $("#error_message").text(e.message);
             $("#error_box").fadeIn("slow");
             $("#login_spinner").hide();
@@ -119,7 +119,7 @@ define(function(require) {
       error_callback = auth_error
     } else {
       two_factor_auth_token = JSON.stringify(publicKeyCredential);
-      error_callback = (req, error) => {
+      error_callback = function(req, error) {
         auth_error(req, error);
         prepareWebAuthn(uid);
       }
diff --git a/src/sunstone/public/app/opennebula/action.js b/src/sunstone/public/app/opennebula/action.js
index 06d1b7935..fd68fa5fc 100644
--- a/src/sunstone/public/app/opennebula/action.js
+++ b/src/sunstone/public/app/opennebula/action.js
@@ -29,7 +29,6 @@ define(function(require) {
 
   var _clearCache = function(cache_name) {
     listCache[cache_name] = null;
-    //console.log(cache_name+" cache cleaned");
   };
 
   //Example: Simple action: publish. Simple action with action obj: deploy
@@ -260,7 +259,6 @@ define(function(require) {
             var callback = listCallbacks[cache_name][i].error;
 
             if (callback) {
-              //console.log(cache_name+" list. ERROR Callback called");
               callback(request, OpenNebulaError(response));
             }
           }
diff --git a/src/sunstone/public/app/opennebula/host.js b/src/sunstone/public/app/opennebula/host.js
index 9c55e4dad..93a7f3219 100644
--- a/src/sunstone/public/app/opennebula/host.js
+++ b/src/sunstone/public/app/opennebula/host.js
@@ -32,7 +32,6 @@ define(function(require) {
 
   var _clearCache = function() {
     infrastructureCache = null;
-    //console.log("Host.pciDevices. Cache cleaned");
   };
 
   var RESOURCE = "HOST";
@@ -145,11 +144,7 @@ define(function(require) {
 
       if (infrastructureCache &&
           infrastructureCache["timestamp"] + CACHE_EXPIRE > new Date().getTime()) {
-
-        //console.log("Host.pciDevices. Cache used");
-
-        return callback ?
-            callback(request, infrastructureCache["pcis"]) : null;
+        return callback ? callback(request, infrastructureCache["pcis"]) : null;
       }
 
       pcisCallbacks.push({
@@ -157,8 +152,6 @@ define(function(require) {
         error : callbackError
       });
 
-      //console.log("Host.pciDevices. Callback queued");
-
       _infrastructure();
     },
     "vcenterCustomizations": function(params){
@@ -168,11 +161,7 @@ define(function(require) {
 
       if (infrastructureCache &&
           infrastructureCache["timestamp"] + CACHE_EXPIRE > new Date().getTime()) {
-
-        //console.log("Host.vcenterCustomizations. Cache used");
-
-        return callback ?
-            callback(request, infrastructureCache["customizations"]) : null;
+        return callback ? callback(request, infrastructureCache["customizations"]) : null;
       }
 
       customizationsCallbacks.push({
@@ -180,8 +169,6 @@ define(function(require) {
         error : callbackError
       });
 
-      //console.log("Host.vcenterCustomizations. Callback queued");
-
       _infrastructure();
     },
     "lxdProfilesInfo": function(params){
@@ -192,8 +179,7 @@ define(function(require) {
       if (infrastructureCache &&
           infrastructureCache["timestamp"] + CACHE_EXPIRE > new Date().getTime()) {
 
-        return callback ?
-            callback(request, infrastructureCache["lxd_profiles"]) : null;
+        return callback ? callback(request, infrastructureCache["lxd_profiles"]) : null;
       }
 
       lxdProfilesInfoCallbacks.push({
@@ -233,8 +219,6 @@ define(function(require) {
 
     infrastructureWaiting = true;
 
-    //console.log("Host.infrastructure. NO cache, calling ajax");
-
     $.ajax({
       url: "infrastructure",
       type: "GET",
@@ -246,7 +230,7 @@ define(function(require) {
           pcis = [];
         }
 
-        if (!$.isArray(pcis)){ // If only 1 convert to array
+        if (!Array.isArray(pcis)){ // If only 1 convert to array
           pcis = [pcis];
         }
 
@@ -256,7 +240,7 @@ define(function(require) {
           customizations = [];
         }
 
-        if (!$.isArray(customizations)){ // If only 1 convert to array
+        if (!Array.isArray(customizations)){ // If only 1 convert to array
           customizations = [customizations];
         }
 
@@ -266,7 +250,7 @@ define(function(require) {
           lxd_profiles = [];
         }
 
-        if (!$.isArray(lxd_profiles)){ // If only 1 convert to array
+        if (!Array.isArray(lxd_profiles)){ // If only 1 convert to array
           lxd_profiles = [lxd_profiles];
         }
 
@@ -276,7 +260,7 @@ define(function(require) {
           kvm_info = [];
         }
 
-        if (!$.isArray(kvm_info)){ // If only 1 convert to array
+        if (!Array.isArray(kvm_info)){ // If only 1 convert to array
           kvm_info = [kvm_info];
         }
 
@@ -294,7 +278,6 @@ define(function(require) {
           var callback = pcisCallbacks[i].success;
 
           if (callback) {
-            //console.log("Host.pciDevices. Callback called");
             callback(request, pcis);
           }
         }
@@ -305,7 +288,6 @@ define(function(require) {
           var callback = customizationsCallbacks[i].success;
 
           if (callback) {
-            //console.log("Host.vcenterCustomizations. Callback called");
             callback(request, customizations);
           }
         }
@@ -342,7 +324,6 @@ define(function(require) {
           var callback = pcisCallbacks[i].error;
 
           if (callback) {
-            //console.log("Host.pciDevices. ERROR Callback called");
             callback(request, OpenNebulaError(response));
           }
         }
@@ -353,7 +334,6 @@ define(function(require) {
           var callback = customizationsCallbacks[i].error;
 
           if (callback) {
-            //console.log("Host.vcenterCustomizations. ERROR Callback called");
             callback(request, OpenNebulaError(response));
           }
         }
diff --git a/src/sunstone/public/app/opennebula/template.js b/src/sunstone/public/app/opennebula/template.js
index ee3143efa..8eab33875 100644
--- a/src/sunstone/public/app/opennebula/template.js
+++ b/src/sunstone/public/app/opennebula/template.js
@@ -148,7 +148,7 @@ define(function(require) {
       if (diskCost != 0) {
         var template_disk = capacity.DISK;
         var disks = [];
-        if ($.isArray(template_disk)) {
+        if (Array.isArray(template_disk)) {
           disks = template_disk;
         } else if (!$.isEmptyObject(template_disk)) {
           disks = [template_disk];
diff --git a/src/sunstone/public/app/opennebula/vm.js b/src/sunstone/public/app/opennebula/vm.js
index 32bef3946..b85be264d 100644
--- a/src/sunstone/public/app/opennebula/vm.js
+++ b/src/sunstone/public/app/opennebula/vm.js
@@ -772,11 +772,11 @@ define(function(require) {
     if (nics == undefined || nics == false){
       nics = [];
     }
-    if (!$.isArray(nics)) {
+    if (!Array.isArray(nics)) {
       nics = [nics];
     }
     if (pci != undefined || pci != false) {
-      if (!$.isArray(pci)) {
+      if (!Array.isArray(pci)) {
         pci = [pci];
       }
       $.each(pci, function(){
diff --git a/src/sunstone/public/app/tabs/clusters-tab/datatable.js b/src/sunstone/public/app/tabs/clusters-tab/datatable.js
index 097e70005..42e050392 100644
--- a/src/sunstone/public/app/tabs/clusters-tab/datatable.js
+++ b/src/sunstone/public/app/tabs/clusters-tab/datatable.js
@@ -114,7 +114,7 @@ define(function(require) {
 
   function _lengthOf(ids){
     var l = 0;
-    if ($.isArray(ids))
+    if (Array.isArray(ids))
       l = ids.length;
     else if (!$.isEmptyObject(ids))
       l = 1;
diff --git a/src/sunstone/public/app/tabs/clusters-tab/panels/datastores.js b/src/sunstone/public/app/tabs/clusters-tab/panels/datastores.js
index f51c7e9e9..bc80bbd1b 100644
--- a/src/sunstone/public/app/tabs/clusters-tab/panels/datastores.js
+++ b/src/sunstone/public/app/tabs/clusters-tab/panels/datastores.js
@@ -60,7 +60,7 @@ define(function(require) {
     if (this.element.DATASTORES.ID != undefined){
       datastores = this.element.DATASTORES.ID;
 
-      if (!$.isArray(datastores)){
+      if (!Array.isArray(datastores)){
         datastores = [datastores];
       }
     }
diff --git a/src/sunstone/public/app/tabs/clusters-tab/panels/hosts.js b/src/sunstone/public/app/tabs/clusters-tab/panels/hosts.js
index 5312f7df3..40dcb7f56 100644
--- a/src/sunstone/public/app/tabs/clusters-tab/panels/hosts.js
+++ b/src/sunstone/public/app/tabs/clusters-tab/panels/hosts.js
@@ -60,7 +60,7 @@ define(function(require) {
     if (this.element.HOSTS.ID != undefined){
       hosts = this.element.HOSTS.ID;
 
-      if (!$.isArray(hosts)){
+      if (!Array.isArray(hosts)){
         hosts = [hosts];
       }
     }
diff --git a/src/sunstone/public/app/tabs/clusters-tab/panels/vnets.js b/src/sunstone/public/app/tabs/clusters-tab/panels/vnets.js
index d685121fd..3f32496bd 100644
--- a/src/sunstone/public/app/tabs/clusters-tab/panels/vnets.js
+++ b/src/sunstone/public/app/tabs/clusters-tab/panels/vnets.js
@@ -60,7 +60,7 @@ define(function(require) {
     if (this.element.VNETS.ID != undefined){
       vnets = this.element.VNETS.ID;
 
-      if (!$.isArray(vnets)){
+      if (!Array.isArray(vnets)){
         vnets = [vnets];
       }
     }
diff --git a/src/sunstone/public/app/tabs/datastores-tab/datatable.js b/src/sunstone/public/app/tabs/datastores-tab/datatable.js
index 55484c890..7e89facae 100644
--- a/src/sunstone/public/app/tabs/datastores-tab/datatable.js
+++ b/src/sunstone/public/app/tabs/datastores-tab/datatable.js
@@ -133,7 +133,7 @@ define(function(require) {
 
     var clusters = '-';
     if (element.CLUSTERS.ID != undefined){
-      clusters = $.isArray(element.CLUSTERS.ID) ? element.CLUSTERS.ID.join(",") : element.CLUSTERS.ID;
+      clusters = Array.isArray(element.CLUSTERS.ID) ? element.CLUSTERS.ID.join(",") : element.CLUSTERS.ID;
     }
 
     var state = OpenNebulaDatastore.stateStr(element.STATE);
diff --git a/src/sunstone/public/app/tabs/datastores-tab/panels/clusters.js b/src/sunstone/public/app/tabs/datastores-tab/panels/clusters.js
index 9c34ec43d..04da1b7ab 100644
--- a/src/sunstone/public/app/tabs/datastores-tab/panels/clusters.js
+++ b/src/sunstone/public/app/tabs/datastores-tab/panels/clusters.js
@@ -59,7 +59,7 @@ define(function(require){
     if (this.element.CLUSTERS.ID != undefined){
       clusters = this.element.CLUSTERS.ID;
 
-      if (!$.isArray(clusters)){
+      if (!Array.isArray(clusters)){
         clusters = [clusters];
       }
     }
diff --git a/src/sunstone/public/app/tabs/datastores-tab/panels/images.js b/src/sunstone/public/app/tabs/datastores-tab/panels/images.js
index 5a5dbc975..52fc89d42 100644
--- a/src/sunstone/public/app/tabs/datastores-tab/panels/images.js
+++ b/src/sunstone/public/app/tabs/datastores-tab/panels/images.js
@@ -65,7 +65,7 @@ define(function(require){
     if (this.element.IMAGES.ID != undefined){
       imgs = this.element.IMAGES.ID;
 
-      if (!$.isArray(imgs)){
+      if (!Array.isArray(imgs)){
         imgs = [imgs];
       }
     }
diff --git a/src/sunstone/public/app/tabs/groups-tab/datatable.js b/src/sunstone/public/app/tabs/groups-tab/datatable.js
index 483313d27..7d23e7e57 100644
--- a/src/sunstone/public/app/tabs/groups-tab/datatable.js
+++ b/src/sunstone/public/app/tabs/groups-tab/datatable.js
@@ -108,7 +108,7 @@ define(function(require) {
     var users_str = "0";
 
     if (element.USERS.ID){
-      if ($.isArray(element.USERS.ID)){
+      if (Array.isArray(element.USERS.ID)){
         users_str = element.USERS.ID.length;
       } else {
         users_str = "1";
diff --git a/src/sunstone/public/app/tabs/groups-tab/panels/users.js b/src/sunstone/public/app/tabs/groups-tab/panels/users.js
index 1d02f0061..2186e47c9 100644
--- a/src/sunstone/public/app/tabs/groups-tab/panels/users.js
+++ b/src/sunstone/public/app/tabs/groups-tab/panels/users.js
@@ -67,7 +67,7 @@ define(function(require) {
     if (this.element.USERS.ID != undefined){
       this.users = this.element.USERS.ID;
 
-      if (!$.isArray(this.users)){
+      if (!Array.isArray(this.users)){
         this.users = [this.users];
       }
     }
@@ -77,7 +77,7 @@ define(function(require) {
     if (this.element.ADMINS.ID != undefined){
       this.admins = this.element.ADMINS.ID;
 
-      if (!$.isArray(this.admins)){
+      if (!Array.isArray(this.admins)){
         this.admins = [this.admins];
       }
     }
diff --git a/src/sunstone/public/app/tabs/hosts-tab/panels/pci.js b/src/sunstone/public/app/tabs/hosts-tab/panels/pci.js
index 9128ea1e9..2b6ef0c38 100644
--- a/src/sunstone/public/app/tabs/hosts-tab/panels/pci.js
+++ b/src/sunstone/public/app/tabs/hosts-tab/panels/pci.js
@@ -76,7 +76,7 @@ define(function(require) {
 
       pcis = this.element.HOST_SHARE.PCI_DEVICES.PCI;
 
-      if (!$.isArray(pcis)){ // If only 1 convert to array
+      if (!Array.isArray(pcis)){ // If only 1 convert to array
         pcis = [pcis];
       }
 
diff --git a/src/sunstone/public/app/tabs/hosts-tab/panels/vms.js b/src/sunstone/public/app/tabs/hosts-tab/panels/vms.js
index d2880fbbd..e182d4a39 100644
--- a/src/sunstone/public/app/tabs/hosts-tab/panels/vms.js
+++ b/src/sunstone/public/app/tabs/hosts-tab/panels/vms.js
@@ -60,7 +60,7 @@ define(function(require){
     if (this.element.VMS.ID != undefined){
       vms = this.element.VMS.ID;
 
-      if (!$.isArray(vms)){
+      if (!Array.isArray(vms)){
         vms = [vms];
       }
     }
diff --git a/src/sunstone/public/app/tabs/hosts-tab/panels/wilds.js b/src/sunstone/public/app/tabs/hosts-tab/panels/wilds.js
index 03b6a7dec..4293c193e 100644
--- a/src/sunstone/public/app/tabs/hosts-tab/panels/wilds.js
+++ b/src/sunstone/public/app/tabs/hosts-tab/panels/wilds.js
@@ -91,7 +91,7 @@ define(function(require) {
 
     if (that.element.TEMPLATE.VM) {
       var wilds = that.element.TEMPLATE.VM;
-      if (!$.isArray(wilds)) { // If only 1 VM convert to array
+      if (!Array.isArray(wilds)) { // If only 1 VM convert to array
         wilds = [wilds];
       }
 
diff --git a/src/sunstone/public/app/tabs/hosts-tab/utils/can-import-wilds.js b/src/sunstone/public/app/tabs/hosts-tab/utils/can-import-wilds.js
index 72f78b48f..185e1ccc2 100644
--- a/src/sunstone/public/app/tabs/hosts-tab/utils/can-import-wilds.js
+++ b/src/sunstone/public/app/tabs/hosts-tab/utils/can-import-wilds.js
@@ -34,7 +34,7 @@ define(function(require){
     var canImportWilds = false;
     if (element.TEMPLATE.VM) {
       var vms = element.TEMPLATE.VM;
-      if (!$.isArray(vms)) { // If only 1 VM convert to array
+      if (!Array.isArray(vms)) { // If only 1 VM convert to array
         vms = [vms];
       }
       $.each(vms, function() {
diff --git a/src/sunstone/public/app/tabs/hosts-tab/utils/datastores-capacity-table.js b/src/sunstone/public/app/tabs/hosts-tab/utils/datastores-capacity-table.js
index 63943afc7..17eae373a 100644
--- a/src/sunstone/public/app/tabs/hosts-tab/utils/datastores-capacity-table.js
+++ b/src/sunstone/public/app/tabs/hosts-tab/utils/datastores-capacity-table.js
@@ -43,7 +43,7 @@ define(function(require){
     var hostShare = element.HOST_SHARE;
 
     var datastores = []
-    if ($.isArray(hostShare.DATASTORES.DS))
+    if (Array.isArray(hostShare.DATASTORES.DS))
       datastores = hostShare.DATASTORES.DS
     else if (!$.isEmptyObject(hostShare.DATASTORES.DS))
       datastores = [hostShare.DATASTORES.DS]
diff --git a/src/sunstone/public/app/tabs/images-tab/panels/snapshots.js b/src/sunstone/public/app/tabs/images-tab/panels/snapshots.js
index f3fbf7da0..6190cca79 100644
--- a/src/sunstone/public/app/tabs/images-tab/panels/snapshots.js
+++ b/src/sunstone/public/app/tabs/images-tab/panels/snapshots.js
@@ -68,7 +68,7 @@ define(function(require){
     if (!$.isEmptyObject(this.element.SNAPSHOTS)){
       var snapshots = this.element.SNAPSHOTS.SNAPSHOT;
 
-      if (!$.isArray(snapshots)){
+      if (!Array.isArray(snapshots)){
         snapshots = [snapshots];
       }
 
diff --git a/src/sunstone/public/app/tabs/images-tab/panels/vms.js b/src/sunstone/public/app/tabs/images-tab/panels/vms.js
index 00fe7baf6..ae2c52494 100644
--- a/src/sunstone/public/app/tabs/images-tab/panels/vms.js
+++ b/src/sunstone/public/app/tabs/images-tab/panels/vms.js
@@ -60,7 +60,7 @@ define(function(require){
     if (this.element.VMS.ID != undefined){
       vms = this.element.VMS.ID;
 
-      if (!$.isArray(vms)){
+      if (!Array.isArray(vms)){
         vms = [vms];
       }
     }
diff --git a/src/sunstone/public/app/tabs/marketplaces-tab/datatable.js b/src/sunstone/public/app/tabs/marketplaces-tab/datatable.js
index d68833427..06eccb844 100644
--- a/src/sunstone/public/app/tabs/marketplaces-tab/datatable.js
+++ b/src/sunstone/public/app/tabs/marketplaces-tab/datatable.js
@@ -159,7 +159,7 @@ define(function(require) {
 
   function _lengthOf(ids){
     var l = 0;
-    if ($.isArray(ids))
+    if (Array.isArray(ids))
       l = ids.length;
     else if (!$.isEmptyObject(ids))
       l = 1;
diff --git a/src/sunstone/public/app/tabs/marketplaces-tab/panels/apps.js b/src/sunstone/public/app/tabs/marketplaces-tab/panels/apps.js
index 30c6ac9fc..71d777bed 100644
--- a/src/sunstone/public/app/tabs/marketplaces-tab/panels/apps.js
+++ b/src/sunstone/public/app/tabs/marketplaces-tab/panels/apps.js
@@ -59,7 +59,7 @@ define(function(require){
     if (this.element.MARKETPLACEAPPS.ID != undefined){
       marketPlaceApps = this.element.MARKETPLACEAPPS.ID;
 
-      if (!$.isArray(marketPlaceApps)){
+      if (!Array.isArray(marketPlaceApps)){
         marketPlaceApps = [marketPlaceApps];
       }
     }
diff --git a/src/sunstone/public/app/tabs/provision-tab/templates/list.js b/src/sunstone/public/app/tabs/provision-tab/templates/list.js
index cdb586386..69cea8ab4 100644
--- a/src/sunstone/public/app/tabs/provision-tab/templates/list.js
+++ b/src/sunstone/public/app/tabs/provision-tab/templates/list.js
@@ -415,7 +415,7 @@ define(function(require) {
 
   function get_provision_disk_image(data) {
     var disks = [];
-    if ($.isArray(data.TEMPLATE.DISK))
+    if (Array.isArray(data.TEMPLATE.DISK))
         disks = data.TEMPLATE.DISK;
     else if (!$.isEmptyObject(data.TEMPLATE.DISK))
         disks = [data.TEMPLATE.DISK];
diff --git a/src/sunstone/public/app/tabs/provision-tab/vms/list.js b/src/sunstone/public/app/tabs/provision-tab/vms/list.js
index 6aff36fae..cc25aa917 100644
--- a/src/sunstone/public/app/tabs/provision-tab/vms/list.js
+++ b/src/sunstone/public/app/tabs/provision-tab/vms/list.js
@@ -1004,7 +1004,7 @@ define(function(require) {
 
   function get_provision_disk_image(data) {
     var disks = [];
-    if ($.isArray(data.TEMPLATE.DISK))
+    if (Array.isArray(data.TEMPLATE.DISK))
         disks = data.TEMPLATE.DISK;
     else if (!$.isEmptyObject(data.TEMPLATE.DISK))
         disks = [data.TEMPLATE.DISK];
diff --git a/src/sunstone/public/app/tabs/secgroups-tab/panels/vms.js b/src/sunstone/public/app/tabs/secgroups-tab/panels/vms.js
index c8c6a9408..4c031d1f3 100644
--- a/src/sunstone/public/app/tabs/secgroups-tab/panels/vms.js
+++ b/src/sunstone/public/app/tabs/secgroups-tab/panels/vms.js
@@ -98,7 +98,7 @@ define(function(require){
     if (attr != undefined){
       ids = attr;
 
-      if (!$.isArray(ids)){
+      if (!Array.isArray(ids)){
         ids = [ids];
       }
     }
diff --git a/src/sunstone/public/app/tabs/settings-tab/panels/group-quotas.js b/src/sunstone/public/app/tabs/settings-tab/panels/group-quotas.js
index 8eba78130..4bb6dc082 100644
--- a/src/sunstone/public/app/tabs/settings-tab/panels/group-quotas.js
+++ b/src/sunstone/public/app/tabs/settings-tab/panels/group-quotas.js
@@ -75,7 +75,7 @@ define(function(require) {
 
     var groups = this.element.GROUPS.ID;
 
-    if (!$.isArray(groups)){
+    if (!Array.isArray(groups)){
       groups = [groups];
     }
 
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/context.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/context.js
index 728bbeed0..d69e4625a 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/context.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/context.js
@@ -256,7 +256,7 @@ define(function(require) {
     var publicClouds = templateJSON["PUBLIC_CLOUD"];
 
     if (publicClouds != undefined) {
-      if (!$.isArray(publicClouds)){
+      if (!Array.isArray(publicClouds)){
         publicClouds = [publicClouds];
       }
 
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/general.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/general.js
index 75fe712c0..62d4c1aff 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/general.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/general.js
@@ -428,7 +428,7 @@ define(function(require) {
       var publicClouds = templateJSON["PUBLIC_CLOUD"];
 
       if (publicClouds != undefined) {
-        if (!$.isArray(publicClouds)){
+        if (!Array.isArray(publicClouds)){
           publicClouds = [publicClouds];
         }
 
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/io.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/io.js
index ee8faf4b4..16cf1b676 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/io.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/io.js
@@ -224,7 +224,7 @@ define(function(require) {
       pcis = [];
     }
 
-    if (!$.isArray(pcis)){ // If only 1 convert to array
+    if (!Array.isArray(pcis)){ // If only 1 convert to array
       pcis = [pcis];
     }
 
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network.js
index 8c4bd1773..dd5f215f9 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network.js
@@ -255,7 +255,7 @@ define(function(require) {
   function _addNicTab(context) {
     var that = this;
 
-    if (!that.nics.find(nic => nic.NAME === ("NIC" + that.numberOfNics))) {
+    if (!that.nics.find(function(nic){ return nic.NAME === ("NIC" + that.numberOfNics) })) {
         that.nics.push({"NAME": "NIC" + that.numberOfNics,
                         "ALIAS": false,
                         "ID": that.numberOfNics,
@@ -291,7 +291,9 @@ define(function(require) {
       var ul = $(this).closest('ul');
       var content = $(target);
       var nicId = content.attr("nicId");
-      var index = that.nics.findIndex(nic => nic.NAME === ("NIC" + (nicId - 1)));
+      var index = that.nics.findIndex(function(nic) {
+        return nic.NAME === ("NIC" + (nicId - 1));
+      })
 
       that.nics.splice(index, 1);
 
@@ -314,7 +316,7 @@ define(function(require) {
     $("#" + LINKS_CONTAINER_ID + " li", context).each(function(index) {
       $("a", this).html(Locale.tr("NIC") + ' ' + index + " <i id='update_remove_nic_" + index + "'class='fa fa-times-circle remove-tab'></i>");
 
-      if (that.nics.find(nic => nic.ALIAS === ("NIC" + index))) {
+      if (that.nics.find(function(nic) { return nic.ALIAS === ("NIC" + index) })) {
         $("#update_remove_nic_" + index).hide();
       } else {
         $("#update_remove_nic_" + index).show();
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network/nic-tab.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network/nic-tab.js
index c82977d1e..17be7d8c4 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network/nic-tab.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/network/nic-tab.js
@@ -518,7 +518,7 @@ define(function(require) {
 
   function _hide_remove(nics) {
     $.each(nics, function(index, value) {
-        if (that.nics.find(nic => nic.ALIAS === value.NAME)) {
+        if (that.nics.find(function(nic) { return nic.ALIAS === value.NAME })) {
             $("#update_remove_nic_" + value.ID).hide();
         } else {
             $("#update_remove_nic_" + value.ID).show();
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/numa.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/numa.js
index a808898fb..5a7768b1a 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/numa.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/numa.js
@@ -87,7 +87,6 @@ define(function(require) {
   }
 
   function successCallback(request, opts, infohost){
-    //console.log("CACHE ", OpenNebulaAction.get_all_cache());
     var selector = $(idsElements.hugepages);
     selector.empty();
     if(infohost && infohost.HOST_POOL && infohost.HOST_POOL.HOST){
@@ -138,8 +137,8 @@ define(function(require) {
     return numaStatus;
   }
 
-  function errorCallback(error, error1){
-    console.log("error->", error, error1);
+  function errorCallback(error){
+    console.log(error);
   }
 
   function _onShow(context, panelForm) {
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/os.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/os.js
index 95898df01..46e2eabb4 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/os.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/create/wizard-tabs/os.js
@@ -442,7 +442,7 @@ define(function(require) {
     if (templateJSON.DISK !== undefined){
       var disks = templateJSON.DISK;
 
-      if (!$.isArray(disks)){
+      if (!Array.isArray(disks)){
         disks = [disks];
       }
       disks = disks.filter(distinct);
@@ -472,7 +472,7 @@ define(function(require) {
     if (templateJSON.NIC !== undefined){
       var nics = templateJSON.NIC;
 
-      if (!$.isArray(nics)){
+      if (!Array.isArray(nics)){
         nics = [nics];
       }
       nics = nics.filter(distinct);
diff --git a/src/sunstone/public/app/tabs/templates-tab/form-panels/instantiate.js b/src/sunstone/public/app/tabs/templates-tab/form-panels/instantiate.js
index 350839299..8b647ed7e 100644
--- a/src/sunstone/public/app/tabs/templates-tab/form-panels/instantiate.js
+++ b/src/sunstone/public/app/tabs/templates-tab/form-panels/instantiate.js
@@ -368,7 +368,7 @@ define(function(require) {
       if(original_tmpl.TEMPLATE.PCI != undefined){
         var original_pcis;
 
-        if ($.isArray(original_tmpl.TEMPLATE.PCI)){
+        if (Array.isArray(original_tmpl.TEMPLATE.PCI)){
           original_pcis = original_tmpl.TEMPLATE.PCI;
         } else if (!$.isEmptyObject(original_tmpl.TEMPLATE.PCI)){
           original_pcis = [original_tmpl.TEMPLATE.PCI];
diff --git a/src/sunstone/public/app/tabs/users-tab/dialogs/groups.js b/src/sunstone/public/app/tabs/users-tab/dialogs/groups.js
index 582e38c85..48afcfc20 100644
--- a/src/sunstone/public/app/tabs/users-tab/dialogs/groups.js
+++ b/src/sunstone/public/app/tabs/users-tab/dialogs/groups.js
@@ -111,7 +111,7 @@ define(function(require) {
     this.element = params.element;
 
     if (this.element.GROUPS !== undefined && this.element.GROUPS.ID !== undefined) {
-      if ($.isArray(this.element.GROUPS.ID)) {
+      if (Array.isArray(this.element.GROUPS.ID)) {
         this.originalGroupIds = this.element.GROUPS.ID;
       } else {
         this.originalGroupIds = [this.element.GROUPS.ID];
diff --git a/src/sunstone/public/app/tabs/users-tab/dialogs/login-token.js b/src/sunstone/public/app/tabs/users-tab/dialogs/login-token.js
index 31ac8acf3..f4e53d903 100644
--- a/src/sunstone/public/app/tabs/users-tab/dialogs/login-token.js
+++ b/src/sunstone/public/app/tabs/users-tab/dialogs/login-token.js
@@ -71,7 +71,7 @@ define(function(require) {
     var tokens = [];
 
     if (this.element != undefined && this.element.LOGIN_TOKEN != undefined){
-      if ($.isArray(this.element.LOGIN_TOKEN)){
+      if (Array.isArray(this.element.LOGIN_TOKEN)){
         tokens = this.element.LOGIN_TOKEN;
       } else {
         tokens = [this.element.LOGIN_TOKEN];
diff --git a/src/sunstone/public/app/tabs/users-tab/dialogs/two-factor-auth.js b/src/sunstone/public/app/tabs/users-tab/dialogs/two-factor-auth.js
index ee2a6e557..ebc60b647 100644
--- a/src/sunstone/public/app/tabs/users-tab/dialogs/two-factor-auth.js
+++ b/src/sunstone/public/app/tabs/users-tab/dialogs/two-factor-auth.js
@@ -139,7 +139,7 @@ define(function(require) {
             return;
           }
           context.on("click", "#add_btn", function () {
-            WebAuthnJSON.create({ "publicKey": response }).then(publicKeyCredential => {
+            WebAuthnJSON.create({ "publicKey": response }).then(function(publicKeyCredential) {
               var nickname = $("#nickname", context).val();
               Sunstone.runAction(
                 "User.enable_sunstone_security_key",
@@ -147,7 +147,7 @@ define(function(require) {
                 { nickname: nickname, publicKeyCredential: publicKeyCredential, current_sunstone_setting: that.sunstone_setting }
               );
             })
-            .catch((e) => {
+            .catch(function(e) {
               Notifier.onError(null, { error: { message: e.message } });
             });              
           });
@@ -166,7 +166,7 @@ define(function(require) {
 
   function randomBase32() {
     const items = Array.from('ABCDEFGHIJKLMNOPQRSTUVWXYZ234567');
-    const result = Array(16).fill(0).map(() => items[Math.floor(Math.random()*items.length)]);
+    const result = Array(16).fill(0).map(function() { return items[Math.floor(Math.random()*items.length)] });
     return result.join('');
   };
 
diff --git a/src/sunstone/public/app/tabs/users-tab/panels/groups.js b/src/sunstone/public/app/tabs/users-tab/panels/groups.js
index 69b9f3175..89ce4d61f 100644
--- a/src/sunstone/public/app/tabs/users-tab/panels/groups.js
+++ b/src/sunstone/public/app/tabs/users-tab/panels/groups.js
@@ -67,7 +67,7 @@ define(function(require) {
     var groups;
     var groupTrHTML = TemplateChgrpTr({"element": this.element});
     if (this.element.GROUPS !== undefined && this.element.GROUPS.ID !== undefined) {
-      if ($.isArray(this.element.GROUPS.ID)) {
+      if (Array.isArray(this.element.GROUPS.ID)) {
         groups = this.element.GROUPS.ID;
       } else {
         groups = [this.element.GROUPS.ID];
diff --git a/src/sunstone/public/app/tabs/vdcs-tab/datatable.js b/src/sunstone/public/app/tabs/vdcs-tab/datatable.js
index 23c0f4814..0665502d9 100644
--- a/src/sunstone/public/app/tabs/vdcs-tab/datatable.js
+++ b/src/sunstone/public/app/tabs/vdcs-tab/datatable.js
@@ -108,7 +108,7 @@ define(function(require) {
     var groupColumn = 0;
 
     var gIds = element.GROUPS.ID;
-    if ($.isArray(gIds)){
+    if (Array.isArray(gIds)){
       groupColumn = gIds.length;
     } else if (!$.isEmptyObject(gIds)){
       groupColumn = 1;
@@ -134,7 +134,7 @@ define(function(require) {
 
   function _lengthOf(ids, res_name){
     var l = 0;
-    if ($.isArray(ids)){
+    if (Array.isArray(ids)){
       l = ids.length;
     } else if (!$.isEmptyObject(ids)){
       if (ids[res_name+"_ID"] == VDC_ALL_RESOURCES){
diff --git a/src/sunstone/public/app/tabs/vdcs-tab/panels/groups.js b/src/sunstone/public/app/tabs/vdcs-tab/panels/groups.js
index bfa774b7e..1d12969f1 100644
--- a/src/sunstone/public/app/tabs/vdcs-tab/panels/groups.js
+++ b/src/sunstone/public/app/tabs/vdcs-tab/panels/groups.js
@@ -60,7 +60,7 @@ define(function(require) {
     if (this.element.GROUPS.ID != undefined){
       groups = this.element.GROUPS.ID;
 
-      if (!$.isArray(groups)){
+      if (!Array.isArray(groups)){
         groups = [groups];
       }
     }
diff --git a/src/sunstone/public/app/tabs/vdcs-tab/utils/common.js b/src/sunstone/public/app/tabs/vdcs-tab/utils/common.js
index 2d2ada13a..526a01391 100644
--- a/src/sunstone/public/app/tabs/vdcs-tab/utils/common.js
+++ b/src/sunstone/public/app/tabs/vdcs-tab/utils/common.js
@@ -43,7 +43,7 @@ define(function(require) {
     if (vdc.CLUSTERS.CLUSTER){
       clusters_array = vdc.CLUSTERS.CLUSTER;
 
-      if (!$.isArray(clusters_array)){
+      if (!Array.isArray(clusters_array)){
         clusters_array = [clusters_array];
       }
     }
@@ -51,7 +51,7 @@ define(function(require) {
     if (vdc.HOSTS.HOST){
       hosts_array = vdc.HOSTS.HOST;
 
-      if (!$.isArray(hosts_array)){
+      if (!Array.isArray(hosts_array)){
         hosts_array = [hosts_array];
       }
     }
@@ -59,7 +59,7 @@ define(function(require) {
     if (vdc.VNETS.VNET){
       vnets_array = vdc.VNETS.VNET;
 
-      if (!$.isArray(vnets_array)){
+      if (!Array.isArray(vnets_array)){
         vnets_array = [vnets_array];
       }
     }
@@ -67,7 +67,7 @@ define(function(require) {
     if (vdc.DATASTORES.DATASTORE){
       datastores_array = vdc.DATASTORES.DATASTORE;
 
-      if (!$.isArray(datastores_array)){
+      if (!Array.isArray(datastores_array)){
         datastores_array = [datastores_array];
       }
     }
diff --git a/src/sunstone/public/app/tabs/vms-tab/actions.js b/src/sunstone/public/app/tabs/vms-tab/actions.js
index 69b5cf50f..57b082c03 100644
--- a/src/sunstone/public/app/tabs/vms-tab/actions.js
+++ b/src/sunstone/public/app/tabs/vms-tab/actions.js
@@ -203,7 +203,7 @@ define(function(require) {
             nics = $.merge(alias, nics)
           }
 
-          var nic = nics.find(n => n.RDP && String(n.RDP).toUpperCase() === "YES");
+          var nic = nics.find(function(n) { return n.RDP && String(n.RDP).toUpperCase() === "YES" });
           var ip = nic && nic.IP ? nic.IP : '';
           var credentials = {};
 
diff --git a/src/sunstone/public/app/tabs/vms-tab/panels/actions.js b/src/sunstone/public/app/tabs/vms-tab/panels/actions.js
index 8e4c32505..b778fa299 100644
--- a/src/sunstone/public/app/tabs/vms-tab/panels/actions.js
+++ b/src/sunstone/public/app/tabs/vms-tab/panels/actions.js
@@ -220,7 +220,7 @@ define(function(require) {
       return empty;
     }
 
-    if (!$.isArray(actions_array)) {
+    if (!Array.isArray(actions_array)) {
       var tmp_array = new Array();
       tmp_array[0]  = actions_array;
       actions_array = tmp_array;
diff --git a/src/sunstone/public/app/tabs/vms-tab/panels/info.js b/src/sunstone/public/app/tabs/vms-tab/panels/info.js
deleted file mode 100644
index eb82445f6..000000000
--- a/src/sunstone/public/app/tabs/vms-tab/panels/info.js
+++ /dev/null
@@ -1,173 +0,0 @@
-/* -------------------------------------------------------------------------- */
-/* Copyright 2002-2020, OpenNebula Project, OpenNebula Systems                */
-/*                                                                            */
-/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
-/* not use this file except in compliance with the License. You may obtain    */
-/* a copy of the License at                                                   */
-/*                                                                            */
-/* http://www.apache.org/licenses/LICENSE-2.0                                 */
-/*                                                                            */
-/* Unless required by applicable law or agreed to in writing, software        */
-/* distributed under the License is distributed on an "AS IS" BASIS,          */
-/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
-/* See the License for the specific language governing permissions and        */
-/* limitations under the License.                                             */
-/* -------------------------------------------------------------------------- */
-
-define(function(require) {
-  /*
-    DEPENDENCIES
-   */
-
-  var Locale = require("utils/locale");
-  var Humanize = require("utils/humanize");
-  var RenameTr = require("utils/panel/rename-tr");
-  var PermissionsTable = require("utils/panel/permissions-table");
-  var TemplateTable = require("utils/panel/template-table");
-  var TemplateTableVcenter = require("utils/panel/template-table");
-  var OpenNebula = require("opennebula");
-  var Navigation = require("utils/navigation");
-
-  /*
-    TEMPLATES
-   */
-
-  var TemplateInfo = require("hbs!./info/html");
-
-  /*
-    CONSTANTS
-   */
-
-  var TAB_ID = require("../tabId");
-  var PANEL_ID = require("./info/panelId");
-  var RESOURCE = "VM";
-  var XML_ROOT = "VM";
-
-  /*
-    CONSTRUCTOR
-   */
-
-  function Panel(info) {
-    this.title = Locale.tr("Info");
-    this.icon = "fa-info-circle";
-    this.element = info[XML_ROOT];
-    return this;
-  }
-
-  Panel.PANEL_ID = PANEL_ID;
-  Panel.prototype.html = _html;
-  Panel.prototype.setup = _setup;
-
-  return Panel;
-
-  /*
-    FUNCTION DEFINITIONS
-   */
-
-  function _html() {
-    var renameTrHTML = RenameTr.html(TAB_ID, RESOURCE, this.element.NAME);
-    var permissionsTableHTML = PermissionsTable.html(TAB_ID, RESOURCE, this.element);
-    var prettyStartTime = Humanize.prettyTime(this.element.STIME);
-
-    var stateStr = OpenNebula.VM.stateStr(this.element.STATE);
-    var lcmStateStr = OpenNebula.VM.lcmStateStr(this.element.LCM_STATE);
-    var hostnameHTML = OpenNebula.VM.hostnameStrLink(this.element);
-    var vrouterHTML = "--";
-
-    var IP = OpenNebula.VM.ipsStr(this.element);
-
-    var alias = (
-      config.system_config &&
-      config.system_config.get_extended_vm_info &&
-      config.system_config.get_extended_vm_info === "true"
-    ) ? null : OpenNebula.VM.aliasStr(this.element);
-
-    if (this.element.TEMPLATE.VROUTER_ID != undefined){
-      vrouterHTML = Navigation.link(
-        OpenNebula.VirtualRouter.getName(this.element.TEMPLATE.VROUTER_ID),
-        "vrouters-tab", this.element.TEMPLATE.VROUTER_ID);
-    }
-
-    var deployId = (typeof(this.element.DEPLOY_ID) == "object" ? "--" : this.element.DEPLOY_ID);
-    var resched = (parseInt(this.element.RESCHED) ? Locale.tr("yes") : Locale.tr("no"));
-
-    // Get rid of the unwanted (for show) SCHED_* keys
-    var that = this;
-    var strippedTemplate = {};
-    var strippedTemplateVcenter = {};
-    var unshownValues = {};
-
-    $.each(that.element.USER_TEMPLATE, function(key, value) {
-      if (key.match(/^SCHED_*/) || key == "USER_INPUTS") {
-        unshownValues[key] = value;
-      }
-      else if (key.match(/^VCENTER_*/)){
-        strippedTemplateVcenter[key] = value;
-      }
-      else {
-        strippedTemplate[key] = value;
-      }
-    });
-
-    var templateTableHTML = TemplateTable.html(strippedTemplate, RESOURCE, Locale.tr("Attributes"), true);
-
-    var templateTableVcenterHTML = TemplateTableVcenter.html(strippedTemplateVcenter, RESOURCE, Locale.tr("vCenter information"), false);
-
-    var monitoring = $.extend({}, this.element.MONITORING);
-    delete monitoring.CPU;
-    delete monitoring.MEMORY;
-    delete monitoring.NETTX;
-    delete monitoring.NETRX;
-    delete monitoring.STATE;
-    delete monitoring.DISK_SIZE;
-    delete monitoring.SNAPSHOT_SIZE;
-    var monitoringTableContentHTML;
-    if (!$.isEmptyObject(monitoring)) {
-      monitoringTableContentHTML = Humanize.prettyPrintJSON(monitoring);
-    }
-
-    return TemplateInfo({
-      "element": this.element,
-      "renameTrHTML": renameTrHTML,
-      "stateStr": stateStr,
-      "lcmStateStr": lcmStateStr,
-      "hostnameHTML": hostnameHTML,
-      "prettyStartTime": prettyStartTime,
-      "deployId": deployId,
-      "IP": IP,
-      "alias": alias,
-      "resched": resched,
-      "permissionsTableHTML": permissionsTableHTML,
-      "templateTableVcenterHTML": templateTableVcenterHTML,
-      "templateTableHTML": templateTableHTML,
-      "monitoringTableContentHTML": monitoringTableContentHTML,
-      "vrouterHTML": vrouterHTML,
-    });
-  }
-
-  function _setup(context) {
-    RenameTr.setup(TAB_ID, RESOURCE, this.element.ID, context);
-    PermissionsTable.setup(TAB_ID, RESOURCE, this.element, context);
-    // Get rid of the unwanted (for show) SCHED_* keys
-    var that = this;
-    var strippedTemplate = {};
-    var strippedTemplateVcenter = {};
-    var unshownValues = {};
-     $.each(that.element.USER_TEMPLATE, function(key, value) {
-      if (key.match(/^SCHED_*/) || key == "USER_INPUTS") {
-        unshownValues[key] = value;
-      }
-      else if (key.match(/^VCENTER_*/)){
-        strippedTemplateVcenter[key] = value;
-      }
-      else {
-        strippedTemplate[key] = value;
-      }
-    });
-    if($.isEmptyObject(strippedTemplateVcenter)){
-      $(".vcenter", context).hide();
-    }
-    TemplateTable.setup(strippedTemplate, RESOURCE, this.element.ID, context, unshownValues, strippedTemplateVcenter);
-    TemplateTableVcenter.setup(strippedTemplateVcenter, RESOURCE, this.element.ID, context, unshownValues, strippedTemplate);
-  }
-});
diff --git a/src/sunstone/public/app/tabs/vms-tab/panels/network.js b/src/sunstone/public/app/tabs/vms-tab/panels/network.js
index 5d7242a05..efcf853b2 100644
--- a/src/sunstone/public/app/tabs/vms-tab/panels/network.js
+++ b/src/sunstone/public/app/tabs/vms-tab/panels/network.js
@@ -253,14 +253,14 @@ define(function(require) {
       return self.indexOf(value)===index;
     };
 
-    if ($.isArray(that.element.TEMPLATE.NIC)){
+    if (Array.isArray(that.element.TEMPLATE.NIC)){
       nics = that.element.TEMPLATE.NIC;
     } else if (!$.isEmptyObject(that.element.TEMPLATE.NIC)){
       nics = [that.element.TEMPLATE.NIC];
     }
 
 
-    if ($.isArray(that.element.TEMPLATE.NIC_ALIAS)){
+    if (Array.isArray(that.element.TEMPLATE.NIC_ALIAS)){
       alias = that.element.TEMPLATE.NIC_ALIAS;
     } else if (!$.isEmptyObject(that.element.TEMPLATE.NIC_ALIAS)){
       alias = [that.element.TEMPLATE.NIC_ALIAS];
@@ -268,7 +268,7 @@ define(function(require) {
 
     var pcis = [];
 
-    if ($.isArray(that.element.TEMPLATE.PCI)){
+    if (Array.isArray(that.element.TEMPLATE.PCI)){
       pcis = that.element.TEMPLATE.PCI;
     } else if (!$.isEmptyObject(that.element.TEMPLATE.PCI)){
       pcis = [that.element.TEMPLATE.PCI];
@@ -334,7 +334,7 @@ define(function(require) {
         var rules = that.element.TEMPLATE.SECURITY_GROUP_RULE;
 
         if (rules != undefined) {
-          if (!$.isArray(rules)) {
+          if (!Array.isArray(rules)) {
             rules = [rules];
           }
 
diff --git a/src/sunstone/public/app/tabs/vms-tab/panels/placement.js b/src/sunstone/public/app/tabs/vms-tab/panels/placement.js
index 1295bf524..94b499fc6 100644
--- a/src/sunstone/public/app/tabs/vms-tab/panels/placement.js
+++ b/src/sunstone/public/app/tabs/vms-tab/panels/placement.js
@@ -80,7 +80,7 @@ define(function(require) {
 
     var history = [];
     if (that.element.HISTORY_RECORDS.HISTORY) {
-      if ($.isArray(that.element.HISTORY_RECORDS.HISTORY))
+      if (Array.isArray(that.element.HISTORY_RECORDS.HISTORY))
           history = that.element.HISTORY_RECORDS.HISTORY;
       else if (that.element.HISTORY_RECORDS.HISTORY.SEQ)
           history = [that.element.HISTORY_RECORDS.HISTORY];
diff --git a/src/sunstone/public/app/tabs/vms-tab/panels/snapshots.js b/src/sunstone/public/app/tabs/vms-tab/panels/snapshots.js
index 3e1aefafe..5b9120e96 100644
--- a/src/sunstone/public/app/tabs/vms-tab/panels/snapshots.js
+++ b/src/sunstone/public/app/tabs/vms-tab/panels/snapshots.js
@@ -94,7 +94,7 @@ define(function(require) {
            <tbody>';
 
     var snapshots = []
-    if ($.isArray(that.element.TEMPLATE.SNAPSHOT))
+    if (Array.isArray(that.element.TEMPLATE.SNAPSHOT))
         snapshots = that.element.TEMPLATE.SNAPSHOT
     else if (!$.isEmptyObject(that.element.TEMPLATE.SNAPSHOT))
         snapshots = [that.element.TEMPLATE.SNAPSHOT]
diff --git a/src/sunstone/public/app/tabs/vms-tab/panels/storage.js b/src/sunstone/public/app/tabs/vms-tab/panels/storage.js
index e10c60a8b..08578a34b 100644
--- a/src/sunstone/public/app/tabs/vms-tab/panels/storage.js
+++ b/src/sunstone/public/app/tabs/vms-tab/panels/storage.js
@@ -181,7 +181,7 @@ define(function(require) {
     var that = this;
 
     var snapshots = [];
-    if ($.isArray(this.element.SNAPSHOTS)){
+    if (Array.isArray(this.element.SNAPSHOTS)){
       snapshots = this.element.SNAPSHOTS;
     } else if (!$.isEmptyObject(this.element.SNAPSHOTS)) {
       snapshots = [this.element.SNAPSHOTS];
@@ -189,7 +189,7 @@ define(function(require) {
 
     var snapshotsSize = {};
     var monitoringSnapshots = [];
-    if ($.isArray(that.element.MONITORING.SNAPSHOT_SIZE))
+    if (Array.isArray(that.element.MONITORING.SNAPSHOT_SIZE))
       monitoringSnapshots = that.element.MONITORING.SNAPSHOT_SIZE;
     else if (!$.isEmptyObject(that.element.MONITORING.SNAPSHOT_SIZE))
       monitoringSnapshots = [that.element.MONITORING.SNAPSHOT_SIZE];
@@ -208,7 +208,7 @@ define(function(require) {
       var diskId = this.DISK_ID;
       var diskSnapshots = this.SNAPSHOT;
 
-      if (!$.isArray(diskSnapshots)){
+      if (!Array.isArray(diskSnapshots)){
         diskSnapshots = [diskSnapshots];
       }
 
@@ -244,7 +244,7 @@ define(function(require) {
     });
 
     var disks = [];
-    if ($.isArray(that.element.TEMPLATE.DISK))
+    if (Array.isArray(that.element.TEMPLATE.DISK))
       disks = that.element.TEMPLATE.DISK.slice(0); // clone
     else if (!$.isEmptyObject(that.element.TEMPLATE.DISK))
       disks = [that.element.TEMPLATE.DISK];
@@ -260,7 +260,7 @@ define(function(require) {
 
     var disksSize = {};
     var monitoringDisks = [];
-    if ($.isArray(that.element.MONITORING.DISK_SIZE))
+    if (Array.isArray(that.element.MONITORING.DISK_SIZE))
       monitoringDisks = that.element.MONITORING.DISK_SIZE;
     else if (!$.isEmptyObject(that.element.MONITORING.DISK_SIZE))
       monitoringDisks = [that.element.MONITORING.DISK_SIZE];
diff --git a/src/sunstone/public/app/tabs/vnets-tab/datatable.js b/src/sunstone/public/app/tabs/vnets-tab/datatable.js
index f33f15ea4..3ffaa457a 100644
--- a/src/sunstone/public/app/tabs/vnets-tab/datatable.js
+++ b/src/sunstone/public/app/tabs/vnets-tab/datatable.js
@@ -127,7 +127,7 @@ define(function(require) {
 
     var clusters = '-';
     if (element.CLUSTERS.ID != undefined){
-      clusters = $.isArray(element.CLUSTERS.ID) ? element.CLUSTERS.ID.join(",") : element.CLUSTERS.ID;
+      clusters = Array.isArray(element.CLUSTERS.ID) ? element.CLUSTERS.ID.join(",") : element.CLUSTERS.ID;
     }
 
     var parent_net = "";
diff --git a/src/sunstone/public/app/tabs/vnets-tab/form-panels/create.js b/src/sunstone/public/app/tabs/vnets-tab/form-panels/create.js
index 577c7f722..059ac3b88 100644
--- a/src/sunstone/public/app/tabs/vnets-tab/form-panels/create.js
+++ b/src/sunstone/public/app/tabs/vnets-tab/form-panels/create.js
@@ -587,7 +587,7 @@ define(function(require) {
         }
       },
       error: function(error){
-        console.log("ERROR", error);
+        console.log(error);
       }
     };
     OpenNebulaAction.list(hostActions,actionHost);
diff --git a/src/sunstone/public/app/tabs/vnets-tab/panels/clusters.js b/src/sunstone/public/app/tabs/vnets-tab/panels/clusters.js
index b9f3ce500..5b0107433 100644
--- a/src/sunstone/public/app/tabs/vnets-tab/panels/clusters.js
+++ b/src/sunstone/public/app/tabs/vnets-tab/panels/clusters.js
@@ -60,7 +60,7 @@ define(function(require){
     if (this.element.CLUSTERS.ID != undefined){
       clusters = this.element.CLUSTERS.ID;
 
-      if (!$.isArray(clusters)){
+      if (!Array.isArray(clusters)){
         clusters = [clusters];
       }
     }
diff --git a/src/sunstone/public/app/tabs/vnets-tab/panels/vrouters.js b/src/sunstone/public/app/tabs/vnets-tab/panels/vrouters.js
index 3924b5dd4..e848de88c 100644
--- a/src/sunstone/public/app/tabs/vnets-tab/panels/vrouters.js
+++ b/src/sunstone/public/app/tabs/vnets-tab/panels/vrouters.js
@@ -60,7 +60,7 @@ define(function(require) {
     if (this.element.VROUTERS.ID != undefined){
       vrs = this.element.VROUTERS.ID;
 
-      if (!$.isArray(vrs)){
+      if (!Array.isArray(vrs)){
         vrs = [vrs];
       }
     }
diff --git a/src/sunstone/public/app/tabs/vnets-templates-tab/panels/clusters.js b/src/sunstone/public/app/tabs/vnets-templates-tab/panels/clusters.js
index 0823eb66c..20924a0e6 100644
--- a/src/sunstone/public/app/tabs/vnets-templates-tab/panels/clusters.js
+++ b/src/sunstone/public/app/tabs/vnets-templates-tab/panels/clusters.js
@@ -60,7 +60,7 @@ define(function(require){
     if (this.element.TEMPLATE.CLUSTERS != undefined){
       clusters = this.element.TEMPLATE.CLUSTER_IDS.split(",");
 
-      if (!$.isArray(clusters)){
+      if (!Array.isArray(clusters)){
         clusters = [clusters];
       }
     }
diff --git a/src/sunstone/public/app/tabs/vnets-topology-tab.js b/src/sunstone/public/app/tabs/vnets-topology-tab.js
index 9d5ccca54..dcce0f87e 100644
--- a/src/sunstone/public/app/tabs/vnets-topology-tab.js
+++ b/src/sunstone/public/app/tabs/vnets-topology-tab.js
@@ -183,7 +183,7 @@ define(function(require) {
       if (vr.VMS.ID != undefined){
         vms = vr.VMS.ID;
 
-        if (!$.isArray(vms)){
+        if (!Array.isArray(vms)){
           vms = [vms];
         }
       }
@@ -231,7 +231,7 @@ define(function(require) {
       if (vnet.VROUTERS.ID != undefined){
         vrs = vnet.VROUTERS.ID;
 
-        if (!$.isArray(vrs)){
+        if (!Array.isArray(vrs)){
           vrs = [vrs];
         }
       }
@@ -535,7 +535,7 @@ define(function(require) {
     if (vr.VMS.ID != undefined){
       vms = vr.VMS.ID;
 
-      if (!$.isArray(vms)){
+      if (!Array.isArray(vms)){
         vms = [vms];
       }
     }
diff --git a/src/sunstone/public/app/tabs/vrouters-tab/panels/info.js b/src/sunstone/public/app/tabs/vrouters-tab/panels/info.js
index e975e127d..0bc1b817a 100644
--- a/src/sunstone/public/app/tabs/vrouters-tab/panels/info.js
+++ b/src/sunstone/public/app/tabs/vrouters-tab/panels/info.js
@@ -76,7 +76,7 @@ define(function(require) {
 
     var nics = [];
 
-    if ($.isArray(this.element.TEMPLATE.NIC)){
+    if (Array.isArray(this.element.TEMPLATE.NIC)){
       nics = this.element.TEMPLATE.NIC;
     } else if (!$.isEmptyObject(this.element.TEMPLATE.NIC)){
       nics = [this.element.TEMPLATE.NIC];
diff --git a/src/sunstone/public/app/tabs/vrouters-tab/panels/vms.js b/src/sunstone/public/app/tabs/vrouters-tab/panels/vms.js
index be471ff78..a33237b7f 100644
--- a/src/sunstone/public/app/tabs/vrouters-tab/panels/vms.js
+++ b/src/sunstone/public/app/tabs/vrouters-tab/panels/vms.js
@@ -60,7 +60,7 @@ define(function(require){
     if (this.element.VMS.ID != undefined){
       vms = this.element.VMS.ID;
 
-      if (!$.isArray(vms)){
+      if (!Array.isArray(vms)){
         vms = [vms];
       }
     }
diff --git a/src/sunstone/public/app/utils/accounting.js b/src/sunstone/public/app/utils/accounting.js
index 6f3075ecc..814146f75 100644
--- a/src/sunstone/public/app/utils/accounting.js
+++ b/src/sunstone/public/app/utils/accounting.js
@@ -494,13 +494,13 @@ define(function(require) {
           // --- DISK ---
 
           var disks = [];
-          if ($.isArray(template.DISK))
+          if (Array.isArray(template.DISK))
             disks = template.DISK;
           else if (!$.isEmptyObject(template.DISK))
             disks = [template.DISK];
 
           var snapshots = [];
-          if ($.isArray(template.SNAPSHOTS)){
+          if (Array.isArray(template.SNAPSHOTS)){
             snapshots = template.SNAPSHOTS;
           } else if (!$.isEmptyObject(template.SNAPSHOTS)) {
             snapshots = [template.SNAPSHOTS];
@@ -514,7 +514,7 @@ define(function(require) {
           // snapshots is an array that contains the snapshots for each DISK
           $.each(snapshots, function(index, snapshotForDisk){
             var diskSnapshots = []
-            if (!$.isArray(snapshotForDisk.SNAPSHOT)){
+            if (!Array.isArray(snapshotForDisk.SNAPSHOT)){
               diskSnapshots = [snapshotForDisk.SNAPSHOT];
             }
 
diff --git a/src/sunstone/public/app/utils/actions.js b/src/sunstone/public/app/utils/actions.js
index d0098f505..632bcf22f 100644
--- a/src/sunstone/public/app/utils/actions.js
+++ b/src/sunstone/public/app/utils/actions.js
@@ -27,7 +27,7 @@ define(function(require) {
         return "";
     }
 
-    if (!$.isArray(actions_array)) {
+    if (!Array.isArray(actions_array)) {
         var tmp_array = new Array();
         tmp_array[0]  = actions_array;
         actions_array = tmp_array;
diff --git a/src/sunstone/public/app/utils/dialogs/clusters.js b/src/sunstone/public/app/utils/dialogs/clusters.js
index 33c6f7e75..960966f85 100644
--- a/src/sunstone/public/app/utils/dialogs/clusters.js
+++ b/src/sunstone/public/app/utils/dialogs/clusters.js
@@ -137,7 +137,7 @@ define(function(require) {
       clusters = this.element.CLUSTERS;
 
       if (clusters !== undefined && clusters.ID !== undefined) {
-        if ($.isArray(clusters.ID)) {
+        if (Array.isArray(clusters.ID)) {
           this.originalClusterIds = clusters.ID;
         } else {
           this.originalClusterIds = [clusters.ID];
@@ -151,7 +151,7 @@ define(function(require) {
 
       if (clusters !== undefined ) {
         clusters = clusters.split(",");
-        if ($.isArray(clusters)) {
+        if (Array.isArray(clusters)) {
           this.originalClusterIds = clusters;
         } else {
           this.originalClusterIds = [clusters];
diff --git a/src/sunstone/public/app/utils/dialogs/generic-confirm.js b/src/sunstone/public/app/utils/dialogs/generic-confirm.js
index 72e6c1929..079b6ba36 100644
--- a/src/sunstone/public/app/utils/dialogs/generic-confirm.js
+++ b/src/sunstone/public/app/utils/dialogs/generic-confirm.js
@@ -72,11 +72,11 @@ define(function(require) {
   function _setParams(params) {
     this.params = params;
 
-    if (this.params.buttons != undefined && !$.isArray(this.params.buttons)){
+    if (this.params.buttons != undefined && !Array.isArray(this.params.buttons)){
       this.params.buttons = [this.params.buttons];
     }
 
-    if (this.params.submit != undefined && !$.isArray(this.params.submit)){
+    if (this.params.submit != undefined && !Array.isArray(this.params.submit)){
       this.params.submit = [this.params.submit];
     }
   }
diff --git a/src/sunstone/public/app/utils/disks-resize.js b/src/sunstone/public/app/utils/disks-resize.js
index 65a41a9c8..ce68d1e7d 100644
--- a/src/sunstone/public/app/utils/disks-resize.js
+++ b/src/sunstone/public/app/utils/disks-resize.js
@@ -73,7 +73,7 @@ define(function(require){
 
     var template_disk = opts.template_json.VMTEMPLATE.TEMPLATE.DISK;
     var disks = [];
-    if ($.isArray(template_disk)) {
+    if (Array.isArray(template_disk)) {
       disks = template_disk;
     } else if (!$.isEmptyObject(template_disk)) {
       disks = [template_disk];
@@ -82,7 +82,7 @@ define(function(require){
     if (opts.template_base_json && opts.template_base_json.VMTEMPLATE) {
       var template_base_disk = opts.template_base_json.VMTEMPLATE.TEMPLATE.DISK;
       var disks_base = [];
-      if ($.isArray(template_base_disk)) {
+      if (Array.isArray(template_base_disk)) {
         disks_base = template_base_disk;
       } else if (!$.isEmptyObject(template_base_disk)) {
         disks_base = [template_base_disk];
diff --git a/src/sunstone/public/app/utils/nics-section.js b/src/sunstone/public/app/utils/nics-section.js
index f9d898c3e..6e8e99eee 100644
--- a/src/sunstone/public/app/utils/nics-section.js
+++ b/src/sunstone/public/app/utils/nics-section.js
@@ -66,7 +66,7 @@ define(function(require) {
 
         var nics = [];
 
-        if ($.isArray(template_nic)){
+        if (Array.isArray(template_nic)){
           nics = template_nic;
         } else if (!$.isEmptyObject(template_nic)){
           nics = [template_nic];
@@ -78,7 +78,7 @@ define(function(require) {
 
         var pcis = [];
 
-        if ($.isArray(template_json.VMTEMPLATE.TEMPLATE.PCI)){
+        if (Array.isArray(template_json.VMTEMPLATE.TEMPLATE.PCI)){
           pcis = template_json.VMTEMPLATE.TEMPLATE.PCI;
         } else if (!$.isEmptyObject(template_json.VMTEMPLATE.TEMPLATE.PCI)){
           pcis = [template_json.VMTEMPLATE.TEMPLATE.PCI];
@@ -93,7 +93,7 @@ define(function(require) {
         var template_alias = template_json.VMTEMPLATE.TEMPLATE.NIC_ALIAS;
         var alias = [];
 
-        if ($.isArray(template_alias)){
+        if (Array.isArray(template_alias)){
           alias = template_alias;
         } else if (!$.isEmptyObject(template_alias)){
           alias = [template_alias];
@@ -115,7 +115,7 @@ define(function(require) {
 
           opt.pci = (nic.TYPE == "NIC");
 
-          if (!_nics.find(nic => nic.NAME === ("NIC" + nicId))) {
+          if (!_nics.find(function(nic) { return nic.NAME === ("NIC" + nicId) })) {
               _nics.push({"NAME": "NIC" + nicId, "ALIAS": false, "DD_ID": provision_nic_accordion_dd_id});
           }
 
@@ -577,7 +577,9 @@ define(function(require) {
     dd_context.on("click", ".provision_remove_nic" , function() {
       dd_context.remove();
 
-      var index = _nics.findIndex(nic => nic.NAME === ("NIC" + dd_context["nic_id"]));
+      var index = _nics.findIndex(function(nic) {
+        return nic.NAME === ("NIC" + dd_context["nic_id"])
+      });
       
       _nics.splice(index, 1);
       
@@ -628,7 +630,7 @@ define(function(require) {
     Foundation.reflow(context, "accordion");
 
     $(".provision_add_network_interface", context).on("click", function() {
-      if (!_nics.find(nic => nic.NAME === ("NIC" + nicId))) {
+      if (!_nics.find(function(nic) { return nic.NAME === ("NIC" + nicId) })) {
           _nics.push({"NAME": "NIC" + nicId, "ALIAS": false, "DD_ID": provision_nic_accordion_dd_id});
       }
 
@@ -696,7 +698,7 @@ define(function(require) {
 
   function _hide_remove() {
     $.each(_nics, function(index, value) {
-        if (_nics.find(nic => nic.ALIAS === value.NAME)) {
+        if (_nics.find(function(nic) { return nic.ALIAS === value.NAME })) {
             $("#remove_nic_" + value.DD_ID).hide();
         } else {
             $("#remove_nic_" + value.DD_ID).show();
diff --git a/src/sunstone/public/app/utils/quotas/quota-defaults.js b/src/sunstone/public/app/utils/quotas/quota-defaults.js
index bfb83e35a..2ea524b4f 100644
--- a/src/sunstone/public/app/utils/quotas/quota-defaults.js
+++ b/src/sunstone/public/app/utils/quotas/quota-defaults.js
@@ -108,7 +108,7 @@ define(function(require) {
 
     var ds_quotas = [];
 
-    if ($.isArray(default_quotas.DATASTORE_QUOTA.DATASTORE))
+    if (Array.isArray(default_quotas.DATASTORE_QUOTA.DATASTORE))
       ds_quotas = default_quotas.DATASTORE_QUOTA.DATASTORE;
     else if (default_quotas.DATASTORE_QUOTA.DATASTORE)
       ds_quotas = [default_quotas.DATASTORE_QUOTA.DATASTORE];
@@ -125,7 +125,7 @@ define(function(require) {
 
     var img_quotas = [];
 
-    if ($.isArray(default_quotas.IMAGE_QUOTA.IMAGE))
+    if (Array.isArray(default_quotas.IMAGE_QUOTA.IMAGE))
       img_quotas = default_quotas.IMAGE_QUOTA.IMAGE;
     else if (default_quotas.IMAGE_QUOTA.IMAGE)
       img_quotas = [default_quotas.IMAGE_QUOTA.IMAGE];
@@ -142,7 +142,7 @@ define(function(require) {
 
     var net_quotas = [];
 
-    if ($.isArray(default_quotas.NETWORK_QUOTA.NETWORK))
+    if (Array.isArray(default_quotas.NETWORK_QUOTA.NETWORK))
       net_quotas = default_quotas.NETWORK_QUOTA.NETWORK;
     else if (default_quotas.NETWORK_QUOTA.NETWORK)
       net_quotas = [default_quotas.NETWORK_QUOTA.NETWORK];
diff --git a/src/sunstone/public/app/utils/quotas/quota-widgets.js b/src/sunstone/public/app/utils/quotas/quota-widgets.js
index 78b89da18..26be157ea 100644
--- a/src/sunstone/public/app/utils/quotas/quota-widgets.js
+++ b/src/sunstone/public/app/utils/quotas/quota-widgets.js
@@ -430,7 +430,7 @@ define(function(require) {
     var img_quotas = [];
 
     if (!empty_quotas){
-      if ($.isArray(info.IMAGE_QUOTA.IMAGE))
+      if (Array.isArray(info.IMAGE_QUOTA.IMAGE))
         img_quotas = info.IMAGE_QUOTA.IMAGE;
       else if (info.IMAGE_QUOTA.IMAGE.ID)
         img_quotas = [info.IMAGE_QUOTA.IMAGE];
@@ -573,7 +573,7 @@ define(function(require) {
     var ds_quotas = [];
 
     if (!empty_quotas){
-      if ($.isArray(info.DATASTORE_QUOTA.DATASTORE))
+      if (Array.isArray(info.DATASTORE_QUOTA.DATASTORE))
         ds_quotas = info.DATASTORE_QUOTA.DATASTORE;
       else if (info.DATASTORE_QUOTA.DATASTORE.ID)
         ds_quotas = [info.DATASTORE_QUOTA.DATASTORE];
@@ -740,7 +740,7 @@ define(function(require) {
     var net_quotas = [];
 
     if (!empty_quotas){
-      if ($.isArray(info.NETWORK_QUOTA.NETWORK))
+      if (Array.isArray(info.NETWORK_QUOTA.NETWORK))
         net_quotas = info.NETWORK_QUOTA.NETWORK;
       else if (info.NETWORK_QUOTA.NETWORK.ID)
         net_quotas = [info.NETWORK_QUOTA.NETWORK];
diff --git a/src/sunstone/public/app/utils/schedule_action.js b/src/sunstone/public/app/utils/schedule_action.js
index 2ee45a7da..ca073db7b 100644
--- a/src/sunstone/public/app/utils/schedule_action.js
+++ b/src/sunstone/public/app/utils/schedule_action.js
@@ -685,7 +685,7 @@ define(function (require) {
       return "";
     }
 
-    if (!$.isArray(actions_array)) {
+    if (!Array.isArray(actions_array)) {
       var tmp_array = new Array();
       tmp_array[0] = actions_array;
       actions_array = tmp_array;
diff --git a/src/sunstone/public/app/utils/spice.js b/src/sunstone/public/app/utils/spice.js
index 9733f0436..cfdb3e410 100644
--- a/src/sunstone/public/app/utils/spice.js
+++ b/src/sunstone/public/app/utils/spice.js
@@ -73,7 +73,6 @@ define(function(require) {
     vm_name = response["vm_name"];
 
     if ((!host) || (!port)) {
-      console.log("must specify host and port in URL");
       return;
     }
 
diff --git a/src/sunstone/public/app/utils/template-utils.js b/src/sunstone/public/app/utils/template-utils.js
index 02ee73736..a21c6f2e3 100644
--- a/src/sunstone/public/app/utils/template-utils.js
+++ b/src/sunstone/public/app/utils/template-utils.js
@@ -66,7 +66,7 @@ define(function(require) {
       } else {
         var values;
 
-        if ($.isArray(value)){
+        if (Array.isArray(value)){
           values = value;
         }else{
           values = [value];
diff --git a/src/sunstone/public/app/utils/vcenter/clusters.js b/src/sunstone/public/app/utils/vcenter/clusters.js
index 554b37283..e7d6060cd 100644
--- a/src/sunstone/public/app/utils/vcenter/clusters.js
+++ b/src/sunstone/public/app/utils/vcenter/clusters.js
@@ -194,7 +194,7 @@ define(function(require) {
         function successClusterList(request, obj_list){
           if (cluster_name && obj_list) {
             let r = null;
-            obj_list.map(cluster => {
+            obj_list.forEach(function(cluster) {
               if (cluster && cluster.CLUSTER && cluster.CLUSTER.NAME) {
                 if (cluster.CLUSTER.NAME === cluster_name){
                   r = cluster.CLUSTER;
@@ -275,8 +275,8 @@ define(function(require) {
           OpenNebulaCluster.list({
             timeout: true,
             success: successClusterList,
-            error: error => {
-              console.log("ERROR", error);
+            error: function(error) {
+              console.log(error);
             }
           });
         } else {
diff --git a/src/sunstone/public/app/utils/vnc.js b/src/sunstone/public/app/utils/vnc.js
index bd93dbfff..167eb745a 100644
--- a/src/sunstone/public/app/utils/vnc.js
+++ b/src/sunstone/public/app/utils/vnc.js
@@ -114,7 +114,6 @@ define(function(require) {
       _rfb.addEventListener("credentialsrequired", credentialsRequired);
     }catch(err){
       setStatus("Something went wrong, connection is closed", "Failed");
-      console.log("error start NOVNC ", err);
     }
 
     $("#open_in_a_new_window").attr("href", link);
